### DISCUSSION TITLE
alternative to Raspberry Pi Pico W?

### DISCUSSION THREAD

**Discussion by MarkuBu** [micropython/micropython#9449] [Hardware & Peripherals]
I'm looking for an alternative for the RaspPi Pico W, maybe a little bit smaller if possible. And it should be available in Germany or at least in the EU.

Any suggestions?

**Comment by teuler** [comment]
If you are looking for a small WiFi enabled board with a rp2040 chip, the Arduino Board NANO RP2040 CONNECT Nano may be an option.

**Reply by MarkuBu** [reply]
Not really because it cost 3 to 4 times as much as the Pico W

**Comment by robert-hh** [comment]
Of ot does not have to be the rp2040, you may look at the seeed xiao esp32c3.

**Reply by MarkuBu** [reply]
Might be an alternative. How portable is MicroPython code from Pico W to the Xiao?

**Reply by steveh127** [reply]
It depends whether you are using a lot of GPIO pins. The XIAO only has 11 GPIO pins versus the 20+ of the Pico which could be a limitation, but it is very much smaller. I use both and the same micropython code runs fine on both. Of course if you are using the Pico PIOs you are stuck with the Pico.

**Comment by scruss** [comment]
What are your requirements?

“small, wifi, cheap and available” are pretty much covered by the [LOLIN](https://lolin.aliexpress.com/store/1331105) boards. I'd avoid the W600 because it's very limited. The D1's ESP8266 can manage many online tasks as long as you're careful with memory. Their ESP32 boards with PSRAM are particularly nice. Things to note:

* The I²C socket looks compatible with Qwiic/Stemma Qt, but isn't. It swaps the data and clock lines.
* Try to buy the official ones (and yes, that AliExpress link is official). Some of the cheaper D1 clones I've tried have had failing build quality.

Portability between different platforms is fair, depending on what hardware-specifics you've use. If you use PIO, it's not portable to anything but RP2.
